app/src/main/java/appricottsoftware/clarity/fragments/ChannelFragment.java

```diff
@@ -273,50 +262,39 @@ private void goToChannelList() {
         searchConstraintLayout.setVisibility(View.INVISIBLE);
         headerConstraintLayout.setVisibility(View.INVISIBLE);
 
-        rListItems = new ArrayList<>();
+        channels = new ArrayList<>();
 
-        int uid = 1;
+        int uid = ClarityApp.getSession(getContext()).getUserID();
         ClarityApp.getRestClient(getContext()).getChannel(uid, new JsonHttpResponseHandler() {
+
             @Override
             public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
                 super.onSuccess(statusCode, headers, response);
-                Log.e(TAG , "onSuccess: Object");
-                //Log.i(TAG, "JSON CHANNEL RESPONSE" + response.toString());
             }
+
             @Override
             public void onSuccess(int statusCode, Header[] headers, JSONArray response) {
-                super.onSuccess(statusCode, headers, response);
+//                super.onSuccess(statusCode, headers, response);
                 Log.e(TAG , "onSuccess: Array");
                 //Log.i(TAG, "JSON CHANNEL RESPONSE" + response.toString());
 
                 try {
                     // Convert the response to Channels
                     TypeToken<ArrayList<Channel>> token = new TypeToken<ArrayList<Channel>>() {};
-                    ArrayList<Channel> channels = ClarityApp.getGson().fromJson(response.toString(), token.getType());
+                    channels = ClarityApp.getGson().fromJson(response.toString(), token.getType());
+                    rAdapter = new ChannelsAdapter(channels, getContext(), true);
+                    channelRecycler.setAdapter(rAdapter);
 
                 } catch(Exception e) {
-                    e.printStackTrace();
+                    Log.e(TAG, "goToChannelList: Failed to get channels", e);
                 }
-
-                try {
-                    for (int i = 0; i < response.length(); i++) {
-                        JSONObject aChannel = response.getJSONObject(i);
-                        String title = aChannel.getString("title").toString();
-                        String imageURL = aChannel.getString("image").toString();
-                        RecyclerListItem newItem = new RecyclerListItem(title, imageURL);
-                        rListItems.add(newItem);
-                    }
-                } catch (JSONException e) {
-                    e.printStackTrace();
-                }
-                rAdapter = new RecyclerAdapter(rListItems, getContext(), true);
-                channelRecycler.setAdapter(rAdapter);
             }
+
             @Override
             public void onSuccess(int statusCode, Header[] headers, String responseString) {
```

Comment:
We don't need a String responseString

Reply:
@mschaffrath I'm removing this block of code.